src/node/types.h

```diff
@@ -99,6 +102,35 @@ struct BlockCheckOptions {
     bool check_pow{true};
 };
 
+struct CoinbaseTemplate {
+    /* nVersion */
+    uint32_t version;
+    /* nSequence for the only coinbase transaction input */
+    uint32_t input_sequence;
+    /**
+     * Bytes which are to be placed at the beginning of scriptSig. Guaranteed
+     * to be less than 8 bytes (not including the length byte). This allows
+     * clients to add up to 92 bytes.
+     */
+    CScript script_sig_prefix;
+    /**
+     * The first (and only) witness stack element of the coinbase.
+     *
+     * Omitted for block templates without witness data.
+     *
+     * This is currently the BIP 141 witness reserved value. A future soft fork
+     * may move the witness reserved value elsewhere, but there will still be a
+     * coinbase witness.
+     */
+    std::optional<uint256> witness;
```

Comment:
In commit "mining: add getCoinbase()" (80acabaf9353245f0839b18d9552ab39f595eda6)

IMO `witness_reserved_value` would be a more descriptive name than just `witness`.